src/i_flmusic.c

```diff
@@ -102,6 +108,78 @@ static fluid_long_long_t FL_sftell(void *handle)
     return mem_ftell((MEMFILE *)handle);
 }
 
+static void AddSoundFont(const char *path)
+{
+    static int size = SOUNDFONTS_INITIAL_SIZE;
+
+    if (soundfonts_num >= size)
+    {
+        size *= 2;
+        soundfonts = I_Realloc(soundfonts, size * sizeof(*soundfonts));
+    }
+
+    soundfonts[soundfonts_num++] = M_StringDuplicate(path);
+}
+
+static void ScanDir(const char *dir)
+{
+    glob_t *glob = I_StartMultiGlob(dir, GLOB_FLAG_NOCASE|GLOB_FLAG_SORTED,
+                                    "*.sf2", "*.sf3", NULL);
+    while(1)
+    {
+        const char *filename = I_NextGlob(glob);
+
+        if (filename == NULL)
+        {
+            break;
+        }
+
+        AddSoundFont(filename);
+    }
+
+    I_EndGlob(glob);
+}
+
+static void GetSoundFonts(void)
+{
+    char *left, *p, *dup_path;
+
+    if (soundfonts_num)
+    {
+        return;
+    }
+
+    soundfonts = malloc(SOUNDFONTS_INITIAL_SIZE * sizeof(*soundfonts));
```

Comment:
This is strictly unnecessary. You could just allocate the array in `AddSoundFont()`, because `realloc(NULL, size)` is the same as `malloc(size)`.

Reply:
This would be `static int size;` and `size = size ? 2 * size : SOUNDFONTS_INITIAL_SIZE` in `AddSoundFont()`. Though that's only cosmetic.

Reply:
Fixed in https://github.com/fabiangreffrath/woof/pull/832/commits/e9a45f3c65dd912eece8353d0086dc82ea8b9d87

Reply:
Thanks!

How about this one?

> A few issues:
> 1. [...]
> 2. If soundfonts are not found, FluidSynth does not appear in the menu.

Reply:
And does it make sense to add this? Is it even a soundfont at all?
`/c/Windows/System32/drivers/gm.dls`

Reply:
`gm.dls` sounds really bad because FluidSynth doesn't emulate Roland SC-55. I think it's better without a soundfont at all.
We can add another exception and show `FluidSynth (no soundfont)`, although this menu item will reset to `module 0 device 0`, this can be even more confusing.

Reply:
Formerly, if we couldn't find the soundfont given by `soundfont_path`, we would show an error message and fall back to Native(Windows)/SDL(elsewhere). Now, I assume we fall back to the first soundfont found in one of the directories given in the `soundfont_dir` variable. 

But what happens now if none is found? Do we still fall back to the default? Do we still show the message?

Reply:
If nosound fonts are found at all and FluidSynth was previously selected, we fallback to module 0 (winmm or sdl), device 0, and display a message. If another music module was previously selected, then FluidSynth silently disappears from the menu.

Reply:
> we fallback to module 0 (winmm or sdl), device 0

Wait, we'll only fall back to the SDL module.

Reply:
But there is no menu entry for SDL on Windows, right?